File: src/main/java/edu/kit/ifv/mobitopp/actitopp/CSVDCParameterLoader.java
package edu.kit.ifv.mobitopp.actitopp;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStream;
import java.io.InputStreamReader;
import java.text.NumberFormat;
import java.text.ParseException;
import java.util.HashMap;
import java.util.Locale;
import java.util.Map;

/**
 * 
 * class to read parameter values for different dc model steps from file system
 * 
 * @author Tim Hilgert
 *
 */
public class CSVDCParameterLoader
{
	  
  /**
   * 
   * method to load parameter values for a specified model step
   * 
   * @param input
   * @param modelstep
   */
  public void loadParameterValues(InputStream input, DCModelSteplnformation modelstep)
  {   
    /*
     * initialization of ModelAlternativeParameterValues objects
     * an object exists for each alternative of this modelstep
     */
    Map<String, DCModelAlternativeParameterValues> alternativesParameters = new HashMap<String, DCModelAlternativeParameterValues>();
    for (String s :  modelstep.getAlternativesList())
    {
    	alternativesParameters.put(s, new DCModelAlternativeParameterValues());
    }
  
		try(BufferedReader inRead = new BufferedReader(new InputStreamReader(input)))
		{
	    boolean header = true;
	    String line = null;

	    while((line  = inRead.readLine())!= null)
	    {
        if(header)
        {
          line = inRead.readLine();
          header = false;
        }
        
        String[] splitted = line.split(";");
        
        String parameterName = splitted[0];
        String alternativeName = splitted[1];
        double parameterValue = 0d;
        NumberFormat nf = NumberFormat.getInstance(Locale.ENGLISH);
        try
        {
          parameterValue = nf.parse(splitted[2]).doubleValue();
        }
        catch (ParseException e)
        {
          e.printStackTrace();
        }
	        
        if(!parameterName.equals("0"))
        {
        	alternativesParameters.get(alternativeName).addParameterValue(parameterName, parameterValue);
        }           
      }
    }
    catch (IOException e)
    {
        e.printStackTrace();
    }
		modelstep.setAlternativesParameters(alternativesParameters);
  }
}

File: src/main/java/edu/kit/ifv/mobitopp/actitopp/CSVDCParameterLoader.kt
package edu.kit.ifv.mobitopp.actitopp

import java.io.BufferedReader
import java.io.IOException
import java.io.InputStream
import java.io.InputStreamReader
import java.text.NumberFormat
import java.text.ParseException
import java.util.Locale

/**
 * class to read parameter values for different dc model steps from file system
 *
 * @author Tim Hilgert
 */
class CSVDCParameterLoader {
    /**
     * method to load parameter values for a specified model step
     *
     * @param input
     * @param modelstep
     */
    fun loadParameterValues(input: InputStream, modelstep: DCModelSteplnformation) {
        /*
         * initialization of ModelAlternativeParameterValues objects
         * an object exists for each alternative of this modelstep
         */
        val alternativesParameters: MutableMap<String, DCModelAlternativeParameterValues> = HashMap()
        for (s in modelstep.alternativesList) {
            alternativesParameters[s] = DCModelAlternativeParameterValues()
        }

        try {
            BufferedReader(InputStreamReader(input)).use { inRead ->
                var header = true
                var line: String?
                while ((inRead.readLine().also { line = it }) != null) {
                    if (header) {
                        line = inRead.readLine()
                        header = false
                    }

                    val splitted = line!!.split(";".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()

                    val parameterName = splitted[0]
                    val alternativeName = splitted[1]
                    var parameterValue = 0.0
                    val nf = NumberFormat.getInstance(Locale.ENGLISH)
                    try {
                        parameterValue = nf.parse(splitted[2]).toDouble()
                    } catch (e: ParseException) {
                        e.printStackTrace()
                    }

                    if (parameterName != "0") {
                        alternativesParameters[alternativeName]!!.addParameterValue(parameterName, parameterValue)
                    }
                }
            }
        } catch (e: IOException) {
            e.printStackTrace()
        }
        modelstep.alternativesParameters = alternativesParameters
    }
}
